pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/AbstractIgnoredAnnotRule.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.rule;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import net.sourceforge.pmd.lang.java.ast.Annotatable;
+import net.sourceforge.pmd.properties.StringMultiProperty;
+
+public abstract class AbstractIgnoredAnnotRule extends AbstractJavaRule {
+
+    protected final StringMultiProperty ignoredAnnotationsDescriptor
```

Comment:
We usually have the property descriptors declared as `static final` constants. That way, it is a bit clearer, that the concrete values, that the property takes, are not stored within the descriptor (because as a static variable, it will be shared between rules/rule instances). The values are stored actually by the rule instance.
Note: the property descriptor name will then change to uppercase: `protected static final StringMultiProperty IGNORED_ANNOTATIONS_DESCRIPTOR = ...`

Reply:
This one uses an abstract (instance) method to populate the default values

Reply:
I think though, that the property could be made private. That way, clients don't build a dependency on it directly

Reply:
ah, ok. Then it can't be static.